accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java

```diff
@@ -5,17 +5,18 @@
 import lombok.AccessLevel;
 import lombok.RequiredArgsConstructor;
 import lombok.experimental.FieldDefaults;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
-@RequestMapping("sessions/")
+@RequestMapping("sessions")
 @RequiredArgsConstructor
 @FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
 public class SessionController {
   AuthorizationService authorizationService;
 
   @PostMapping("login")
-  String login(@RequestBody LoginPayload payload) {
+  ResponseEntity<String> login(@RequestBody LoginPayload payload) {
```

Comment:
add validation here, at least @NotNull annotations in your entity

Reply:
Added **@Validated** annotation to enforce validation rules for @NotBlank and @NotNull annotations.

Reply:
A new commons-validation package has been created, which includes a GlobalExceptionHandler that can handle MethodArgumentNotValidException globally in the application